config/config_test.go

```diff
@@ -58,3 +62,42 @@ func TestDurationMarshalJSON(t *testing.T) {
 	assert.Equal(t, `"2s"`, string(data))
 	assert.NoError(t, err)
 }
+
+func TestFromConfig(t *testing.T) {
+
+	timeout := 20 * time.Second
+	backend := fiber.NewBackend("proxy_name", "localhost:1234")
+
+	httpDispatcher, _ := fiberhttp.NewDispatcher(&http.Client{Timeout: timeout})
+	httpCaller, _ := fiber.NewCaller("proxy_name", httpDispatcher)
+	httpProxy := fiber.NewProxy(backend, httpCaller)
+
+	grpcDispatcher := &grpc.Dispatcher{Timeout: timeout}
+	grpcCaller, _ := fiber.NewCaller("proxy_name", grpcDispatcher)
+	grpcProxy := fiber.NewProxy(backend, grpcCaller)
+
+	tests := []struct {
+		name       string
+		configPath string
+		want       fiber.Component
+	}{
+		{
+			name:       "http proxy",
+			configPath: "../internal/testdata/config/http_proxy.yaml",
+			want:       httpProxy,
+		},
+		{
+			name:       "grpc proxy",
+			configPath: "../internal/testdata/config/grpc_proxy.yaml",
+			want:       grpcProxy,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := config.FromConfig(tt.configPath)
```

Comment:
i agree, will see how can i move these into the component, since we are creating the component afterall.

Reply:
for now i renamed it to `InitComponentFromConfig`, since it executes the interface method of the config.

Also, more than half the components (Dispatcher, Fan in, backend etc) doesn't supported creation from config. There are other config (strategy, routes) which are not components but also within this file which don't implement the interface `initComponent` either. So moving `FromConfig` to component won't work. 

Essentially, this only works for `eager_router` or `lazy_router` and opinionated (routes will always be created with proxy, caller, backend and dispatcher. It could be other composition). And probably shouldn't be in fiber root package.

I suggest do this refactoring in another MR as it isn't trivia and re-structuring this (separate them as different file under the same package etc) would be more helpful. Ideally every component should be able to generate from config and the component generation should be more explicit (what are the child components composition). This would fit a generic use-case better.